### PR TITLE
web binding beta

### PR DESCRIPTION
**Pull request by ksyeo1010**
- add leopard web binding

### CODE REVIEW

**Review by mrrostam — APPROVED** [review]

File: binding/web/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+```console
+npx pvbase64 -i ${LEOPARD_MODEL_FILE} -o ${OUTPUT_DIRECTORY}/${MODEL_NAME}.js
+```
+
+The output will be a js file which you can import into any file of your project. For detailed information about `pvbase64`,
+run:
+
+```console
+npx pvbase64 -h
```
Comment (mrrostam) [inline binding/web/README.md:72]: this package has not released yet, right?

File: binding/web/src/leopard.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+type pv_status_to_string_type = (status: number) => Promise<number>
+type pv_sample_rate_type = () => Promise<number>;
+type pv_leopard_version_type = () => Promise<number>;
+
+/**
+* JavaScript/WebAssembly Binding for Leopard
+*/
+
+type LeopardWasmOutput = {
+  aligned_alloc: CallableFunction;
```
Comment (mrrostam) [inline binding/web/src/leopard.ts:43]: why not "aligned_alloc_type"?

Comment (mrrostam) [inline binding/web/src/leopard.ts:43]: why not "aligned_alloc_type"

File: binding/web/src/leopard.ts [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    const pv_free = exports.pv_free as pv_free_type;
+    const pv_leopard_version = exports.pv_leopard_version as pv_leopard_version_type;
+    const pv_leopard_process = exports.pv_leopard_process as pv_leopard_process_type;
+    const pv_leopard_delete = exports.pv_leopard_delete as pv_leopard_delete_type;
+    const pv_leopard_init = exports.pv_leopard_init as pv_leopard_init_type;
+    const pv_status_to_string = exports.pv_status_to_string as pv_status_to_string_type;
+    const pv_sample_rate = exports.pv_sample_rate as pv_sample_rate_type;
+
+    const transcriptionAddressAddress = await aligned_alloc(
+      Uint8Array.BYTES_PER_ELEMENT,
```
Comment (mrrostam) [inline binding/web/src/leopard.ts:308]: i think this needs to be "Int32Array" as it's a pointer

Comment (mrrostam) [inline binding/web/src/leopard.ts:308]: i think this needs to be "Int32Array" as it's a pointer

File: binding/web/src/leopard_worker_handler.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      }
+      break;
+    case 'process':
+      // eslint-disable-next-line no-case-declarations
+      const transferable = (event.data.transfer) ? [event.data.pcm.buffer] : [];
+      if (leopard === null) {
+        self.postMessage({
+          command: "failed",
+          message: "Leopard not initialized"
+        }, transferable);
```
Comment (mrrostam) [inline binding/web/src/leopard_worker_handler.ts:56]: why do we need this here?

Comment (mrrostam) [inline binding/web/src/leopard_worker_handler.ts:56]: why do we need this here?

File: binding/web/README.md (hunk 1 above)
Comment (ksyeo1010) [reply binding/web/README.md:72]: it is -> comes with web utils

File: binding/web/README.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+### Usage
+
+#### Init options
+
+Leopard saves and caches your model file in IndexedDB to be used by Web Assembly. Use a different `modelPath` variable
```
Comment (kenarsa) [inline binding/web/README.md:79]: Web Assembly >>> WebAssembly

File: binding/web/src/leopard_worker_handler.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+ */
+let leopard: Leopard | null = null;
+self.onmessage = async function (
+  event: MessageEvent<LeopardWorkerRequest>
+): Promise<void> {
+  switch (event.data.command) {
+    case 'init':
+      if (leopard !== null) {
+        self.postMessage({
+          command: "failed",
```
Comment (kenarsa) [inline binding/web/src/leopard_worker_handler.ts:29]: should we treat this as a failure?

File: binding/web/src/leopard_worker_handler.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+          message: "Leopard already initialized"
+        });
+        return;
+      }
+      Leopard.setWasm(event.data.wasm);
```
Comment (kenarsa) [inline binding/web/src/leopard_worker_handler.ts:34]: should we put this inside `try`?

**Pull request merged**